public_html/js/views/login.js

```diff
@@ -28,25 +30,34 @@ define([
         },
 
         submit: function(e) {
+
             e.preventDefault();
 
+            var $this = this;
+
             var username = $('#username').val();
             var password = $('#password').val();
 
-            if ( session.validateLogin(username, password) ) {
+            var valid = session.validateLogin(username, password)
 
-                session.login(username, password);
-                $(window).ajaxError(function() {
-                        $('.form__error').hide();
-                        $('.js-login-error').show();
-                });
-                $(window).ajaxSuccess(
-                    function() {
-                        Backbone.history.navigate('game', { trigger: true })
-                });
+            if ( valid === 'None' ) {
+
+                session.login(username, password)
+                    .done(function() {
+                        Backbone.history.navigate('game', { trigger: true });
+                    })
+                    .fail(function(){
+                        $this.$el.find('.form__error').hide();
+                        $this.$el.find('.form__login__error').show();
+                    });
 
             } else {
-                 this.$el.find('.js-username-error, .js-password-error').text('Required').show();
+
+                this.$el.find('.form__error').hide();
+                valid.forEach(function (item) {
+                    $('.form__'+ item +'__error').text("Required").show()
```

Comment:
Добрый день, посмотрите пожалуйста изменения в реквесте)

2016-03-29 13:38 GMT+03:00 Egor Dydykin <notifications@github.com>:

> In public_html/js/views/login.js
> <https://github.com/frontend-park-mail-ru/2016_1_ShockWork/pull/9#discussion_r57702934>
> :
>
> >
> >              } else {
> > -                 this.$el.find('.js-username-error, .js-password-error').text('Required').show();
> > +
> > +                this.$el.find('.form__error').hide();
> > +                valid.forEach(function (item) {
> > +                    $('.form__'+ item +'__error').text("Required").show()
>
> я вижу иное, если ошибок нет, идем логинится, если любые ошибки есть,
> показываем всем итемам с ошибками required
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/frontend-park-mail-ru/2016_1_ShockWork/pull/9/files/cbdf59a63d24857b0e0d7c215e02ff993dc94c1e#r57702934>
>